cmd/private-org-sync/main.go

```diff
@@ -398,6 +392,67 @@ func (g gitSyncer) mirror(repoDir string, src, dst location) error {
 	return nil
 }
 
+func addGitRemote(logger *logrus.Entry, git gitFunc, org, repo, repoDir, remoteName string) error {
+	remoteSetupLogger := logger.WithField("remote-name", remoteName)
+	remoteSetupLogger.Debug("Remote does not exist, setting up")
+
+	srcUrl, err := url.Parse(fmt.Sprintf("https://github.com/%s/%s", org, repo))
+	if err != nil {
+		remoteSetupLogger.WithError(err).Error("Failed to construct URL for the source remote")
+		return fmt.Errorf("failed to construct URL for the source remote")
+	}
+
+	remoteSetupLogger = remoteSetupLogger.WithField("remote-url", srcUrl.String())
+	remoteSetupLogger.Debug("Adding remote")
+
+	if _, exitCode, err := git(logger, repoDir, "remote", "add", remoteName, srcUrl.String()); err != nil || exitCode != 0 {
+		remoteSetupLogger.WithField("exit-code", exitCode).WithError(err).Error("Failed to set up source remote")
+		return fmt.Errorf("failed to set up source remote")
+	}
+
+	return nil
+}
+
+func mergeRemotesAndPush(logger *logrus.Entry, git gitFunc, repoDir, srcRemote, destRemote, org, repo, branch, destUrl string, confirm bool) error {
+	if err := addGitRemote(logger, git, org, repo, repoDir, destRemote); err != nil {
```

Comment:
There's no need to make a named remote for this: we can easily use the URL instead in the `fetch/push` commands. We already built the URL in the `mirror` method, in the `destUrl` variable.

And if we do not need to add a named remote, we do not need `addGitRemote` as a new, separate method ;)